ctapipe/io/containers.py

```diff
@@ -78,6 +77,7 @@ class RawCameraContainer(Container):
                             "integrated ADC data (n_pixels)"))
     adc_samples = Item(Map(), ("map of channel to arrays of "
                                "(n_pixels, n_samples)"))
+    num_samples = Item(int, "number of time samples")
```

Comment:
remember that anything in this container is something that changes *per event*, so `num_samples` does not belong here. It could however go in the container's metadata.  I would still call it an instrumental feature, though, so it should be in the `inst` container.